builq.go

```diff
@@ -45,8 +46,28 @@ func (b *Builder) Addf(format string, args ...any) *Builder {
 	return b
 }
 
+var missingRE = regexp.MustCompile(`%!.\(MISSING\)`)
+
 func (b *Builder) Build() (string, []any, error) {
-	return b.query.String(), b.args, b.err
+	// prioritize the errors from the query string,
+	// otherwise Build might return a wrong error.
+
+	query := b.query.String()
+	switch {
+	case missingRE.MatchString(query):
+		return "", nil, errTooFewArguments
+	case strings.Contains(query, "%!(EXTRA"):
+		return "", nil, errTooManyArguments
+	case strings.Contains(query, "(PANIC="):
+		// TODO(junk1tm): investigate panic cases
+		// 	_, msg, _ := strings.Cut(query, "(PANIC=")
+		// 	return "", nil, errors.New(msg)
+	}
+
+	if b.err != nil {
+		return "", nil, b.err
+	}
+	return query, b.args, nil
```

Comment:
Should we do `return b.query.String(), b.args, b.err` all the time so the query can be logged printed even with an error? due to hidden/inplace fmt error this sounds like a good idea, wdyt?

Reply:
It might be unexpected because usually when `err != nil`, all other return values are zero. Also the real life usage of `Build` is the following:
```go
query, args, err := b.Build()
if err != nil {
    // no interaction with the query
    return err
}
```
Not sure if there is a use case for loggin the query 🤔

Reply:
I'm about debug like:

```
query, args, err := b.Build()
if err != nil {
    println("wtf", query)
    return err
}
```

I see more benefits to return query rather than ""

Reply:
Noted, rewrote with `b.err` reassigning